[Dai Tianle] iP

ddx-510 opened the pull request:
### DDXTodo

> Your time is limited, so don't waste it living someone else's life. - [Steve Jobs](https://news.stanford.edu/2005/06/14/jobs-061505/#:~:text=Your%20time%20is%20limited%2C%20so,follow%20your%20heart%20and%20intuition.)

**DDXTodo** frees your mind of having to remember things you need to do and helps you live a more efficient life.

 It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/ddx-510/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 

And it is FREE!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines
- [x]  Event Reminders
- [ ] Categorize tasks (Coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Level-1 increment

Commit:
Level-2 increment

Commit:
Level-3 increment

Commit:
Level-4 increment

Commit:
A-TextUiTesing increment

Commit:
Level-5 increment

Commit:
Level-6 increment

Commit:
A-Enums implemented

Commit:
Update to increment level-7

Commit:
Update localDateTime to Deadline and Event classes.

Commit:
Merge branch 'branch-Level-8'

Commit:
Fix local datetime in string parser.

Commit:
Restruct the whole project into orgainsed oop.

Commit:
Divide classes into packages.

Commit:
Add Junit Test to duke.

Commit:
Package the app as a JAR file.

Commit:
Update to level-9 increment with implementation of find by keyword.

Commit:
Merge branch 'branch-Level-9'

Commit:
Add java doc for classes and methods.

Commit:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/helpTool/Ui.java

Commit:
Fix missing java doc.

Commit:
Tweak the code to comply with a coding standard.

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Fix merging issues.

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

Commit:
Merge branch 'A-Gradle'

Commit:
Add checkstyle file for gradle

Commit:
Update code to meet checkstyle

Commit:
Merge branch 'A-CheckStyle'

Commit:
Modify the chatbot to GUI and update related methods

Commit:
Update GUI to fxml

Commit:
Update checkstyle and javadoc

Commit:
Update css to fxml file

Commit:
Merge branch 'branch-Level-10'

reignnz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+     * @return the response
+     */
+    public String getResponse(String input) {
+        String tempResult = "";
```
Should this be named result?

ddx-510 replied:
Noted

reignnz commented on src/main/java/duke/helptool/Ui.java:
```diff
@@ -0,0 +1,120 @@
+    /**
+     * Bye.
+     */
+    public String bye() {
```
Should this be showBye?

reignnz commented on src/main/java/duke/helptool/Parser.java:
```diff
@@ -0,0 +1,159 @@
+ * The type Parser.
+ */
+public class Parser {
+    private static final DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
I like how you included the dateTimeFormatter here instead of at the print statement

reignnz commented on src/main/java/duke/helptool/TaskList.java:
```diff
@@ -0,0 +1,69 @@
+     * @param arrayList the array list
+     */
+    public TaskList(ArrayList<Task> arrayList) {
+        this.taskList = Objects.requireNonNullElseGet(arrayList, ArrayList::new);
```
I like how you made use of a method within the Object class to check if the arrayList parameter is null or not

ddx-510 replied:
Thanks :)

reignnz commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,33 @@
+    /**
+     * The At date.
+     */
+    protected LocalDateTime at;
```
Should the variable name be atDateTime?

ddx-510 replied:
Noted, I will change that.

reignnz reviewed (commented):
LGTM.

Commit:
Implement assertions

Currently no assertions are used to ensure certain assumptions hold.

Need to include the assertion so that the code will not crash when
certain assumptions are missed, and inform the author of the code, which
assumptions are violated.

Let's include assertion in different parts of the code:
- AddCommand.java : assert to ensure exceptions from adding tasks are
  caught
- DeleteCommand.java : assert to ensure delete exceptions are caught
- FindCommand.java : assert to ensure find details are not empty
- MarkCommand.java : assert to ensure exceptions are caught
- UnmarkCommand.java : assert to ensure exceptions are caught
- TaskList.java : assert to ensure get and delete operation are of
  correct index
- Task.java : assert to ensure task description is not null

Beyond assertion, method names are optimized to comply with naming
standards.

Commit:
Improve code quality

Currently most of the codes are complied with coding standard, but there
are a few more lines that is not of standard level.

With code quality below standard, it is hard for others to read and
understand the code, it is even hard for the author, to maintain and
modify the code.

Let's improve the code quality on following pieces of code:
- ExitCommand.java: adjust method name to start with verb
- Parser.java: adjust the if else statement to avoid deep nesting
- Ui.java: adjust method nmae to start with verb
Those modifications could help improve code readability and code
quality.

The detailed standard can be found here:
https://nus-cs2103-ay2122s2.github.io/website/schedule/week5/topics.html#readability

Commit:
Merge pull request #1 from ddx-510/branch-A-Assertions

Implement assertions

Commit:
Merge pull request #2 from ddx-510/branch-A-CodeQuality

Improve code quality

Commit:
Create gradle.yml

Commit:
Remove extra I/O redirection tests

Commit:
Add Tag feature to Task, AddCommand, parser and storage

Commit:
Fix gradle test error

Commit:
Fix checkstyle error

Commit:
Merge pull request #3 from ddx-510/branch-C-Tagging

Add Tag feature to Task, AddCommand, parser and storage

Commit:
Update README.md

HanJiyao commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,26 @@
+/**
+ * The type Command.
+ */
+public abstract class Command {
+    /**
+     * Is exit boolean.
```
I like how you separated command using abstract class

HanJiyao commented on src/main/java/duke/helptool/Parser.java:
```diff
@@ -0,0 +1,225 @@
+                if (input.startsWith("mark")) {
+                    // detect digit
+                    if (isDigit) {
+                        throw new DukeException("☹ OOPS!!! The mark cannot be done.");
```
maybe can say why it cannot be done e.g. must be digit

HanJiyao commented on src/main/java/duke/helptool/Storage.java:
```diff
@@ -0,0 +1,138 @@
+     * @throws DukeException the duke exception
+     */
+    public void create() throws DukeException {
+        try {
+            File myObj = new File(this.filePath);
+            boolean isCreated = myObj.createNewFile();
```
how about the file path directory? maybe can consider mkdir() if not exist

HanJiyao commented on src/test/java/duke/helptool/TaskListTest.java:
```diff
@@ -0,0 +1,44 @@
+
+    @Test
+    @DisplayName("Determine size should work")
+    void testTaskListSize() {
```
the convention is it should be get_tasklist_size to be more clear?

HanJiyao commented on src/main/java/duke/tag/Tag.java:
```diff
@@ -0,0 +1,31 @@
+package duke.tag;
+
+/**
+ * The type Tag.
+ */
```
I like the new function of tag

HanJiyao commented on src/main/java/duke/helptool/Storage.java:
```diff
@@ -0,0 +1,138 @@
+            return tempTaskList;
+        } catch (FileNotFoundException e) {
+            this.create();
+            return null;
```
is there better to have exception message to return?

HanJiyao reviewed (commented):
Overall, I found your code easy to read and clear for the most part. And well done that I did not noted coding standard violations. In addition good coding style make the orgainzation and JavaDoc very clear. Just some minor suggestions.

Commit:
Fix bugs of empty tags and update new GUI

Commit:
Merge branch 'master' of https://github.com/ddx-510/ip

Commit:
Fix code standard issues

Commit:
Set theme jekyll-theme-cayman

Commit:
Update ui.png and user guide

Commit:
Merge branch 'master' of https://github.com/ddx-510/ip

Commit:
Improve user guide format

Commit:
Change main class to launcher
